### PR TITLE
[TASK] Add tests for `RuleValueList::getArrayRepresentation()`

### PR DESCRIPTION
**Pull request by oliverklee**
Part of #1440.

### CODE REVIEW

File: tests/Unit/Value/Fixtures/ConcreteRuleValueList.php [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Sabberworm\CSS\Tests\Unit\Value\Fixtures;
+
+use Sabberworm\CSS\Value\RuleValueList;
+
+final class ConcreteRuleValueList extends RuleValueList {}
```
Comment (JakeQZ) [inline tests/Unit/Value/Fixtures/ConcreteRuleValueList.php]: `RuleValueList` is already a concrete class.  This is not necessary.

**Review by JakeQZ — CHANGES_REQUESTED** [review]
We don't need `ConcreteRuleValueList` as `RuleValueList` is already a concrete class.

File: tests/Unit/Value/Fixtures/ConcreteRuleValueList.php (hunk 1 above)
Comment (oliverklee) [reply tests/Unit/Value/Fixtures/ConcreteRuleValueList.php]: Oh, indeed. It seems I got confused by the class also getting subclassed. Fixed and repushed.

File: tests/Unit/Value/RuleValueListTest.php [hunk 2]
```diff
@@ -25,4 +26,60 @@ public function getArrayRepresentationIncludesClassName(): void
...
+        self::assertSame('+', $result['components'][1]['value']);
+        self::assertSame('Size', $result['components'][2]['class']);
+    }
+
+    /**
+     * @test
+     */
+    public function getArrayRepresentationIncludesSeparator(): void
+    {
+        $separator = ', ';
```
Comment (JakeQZ) [inline tests/Unit/Value/RuleValueListTest.php:78]: We should probably add a test that the default separator is returned if none is provided to the constructor.  And in `ValueListTest` also check that a non-default separator is returned.  But that can be for another PR.

**Review by JakeQZ — APPROVED** [review]
The implementation was put in to the abstract class `ValueList`, and tests were added for that class.

These new tests are for the concrete subclass `RuleValueList`, and pretty much the same, to confirm that nothing is broken by the subclassing.
